crates/daphne/src/protocol/mod.rs

```diff
@@ -1,23 +1,76 @@
 // Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
 // SPDX-License-Identifier: BSD-3-Clause
 
+use prio::codec::{CodecError, Decode as _};
+use std::{collections::HashSet, io::Cursor};
+
 pub(crate) mod aggregator;
 mod client;
 mod collector;
+pub(crate) mod report_init;
+
+/// checks if an iterator has no duplicate items, returns the ok if there are no dups or an error
+/// with the first offending item.
+fn no_duplicates<I>(iterator: I) -> Result<(), I::Item>
+where
+    I: Iterator,
+    I::Item: Eq + std::hash::Hash + Copy,
```

Comment:
is it possible to remove the `Copy` constraint?

Reply:
Removed